### PR TITLE
Feature/main&notifications add property

### PR DESCRIPTION
**Pull request by rudenkoserhii**
## Describe your changes

- I added linked from page with PropertiesList page property/[id] with detailed information about property. Added slider for images and modal with location.

## Issue ticket code (and/or) and link

- [[Link to JIRA ticket](https://testhomework4.atlassian.net/browse/AG-126?atlOrigin=eyJpIjoiYzc0NWY0NGMxYWNlNGQxMDg2YTNjYzIxYjQ1ZmM0ZGMiLCJwIjoiaiJ9)]

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [ ] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [ ] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/custom/property/components/InfoBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  TypographyDescriptionRight,
+  TypographyInsert,
+} from '../styles';
+import { fCurrency } from 'utils/format-number';
+import { useTranslations } from 'next-intl';
+import { IPropertyDetailed } from 'types/property';
+import { getCountries } from 'sections/verification-view/drop-box-data';
+import splitValue from '../helpers/splitValue';
+import firstUpperCase from '../helpers/firstUpperCase';
+
```
Comment (anette1983) [inline src/components/custom/property/components/InfoBlock.tsx:16]: тут треба виправити порядок імпортів

File: src/components/custom/property/components/InfoBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  return (
+    <Box
+      sx={{
+        borderEndEndRadius: '8px',
+        borderEndStartRadius: '8px',
+        borderLeft: '1px solid #00a76f',
+        borderRight: '1px solid #00a76f',
+        borderBottom: '1px solid #00a76f',
+        p: '1.5rem',
+      }}
```
Comment (anette1983) [inline src/components/custom/property/components/InfoBlock.tsx:70]: Кольори мають бути у змінних або у темі

File: src/components/custom/property/index.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import { useSnackbar } from 'components/snackbar';
+import SlickSlider from './components/SlickSlider';
+import getImages from './helpers/getImages';
+import ViewOnMap from './components/ViewOnMap';
+import Iconify from 'components/iconify';
+import InfoBlock from './components/InfoBlock';
+import { Title, TypographyStyled, ButtonStyled, Wrapper } from './styles';
+import Image from 'components/image/image';
+import useScrollToTop from '../propertiesList/hooks/useScrollToTop';
+
```
Comment (anette1983) [inline src/components/custom/property/index.tsx:21]: порядок імпортів

File: src/components/custom/property/index.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+const DEFAULT_IMAGE = '/assets/images/home/properties_blank.jpg';
+
+export default function Property({ id }: { id: string }): JSX.Element {
+  const { property, propertyError } = useGetProperty(id);
+  const [propertyDetailed, setPropertyDetailed] = useState<IPropertyDetailed>();
+  const [error, setError] = useState<AxiosError>();
+  const [openModal, setOpenModal] = useState(false);
+
```
Comment (anette1983) [inline src/components/custom/property/index.tsx:29]: тип `useState<boolean>`

**Comment by anette1983** [comment]
Сергію, додай себе як Assignee до пул реквесту і додай якийсь Label

**Comment by rudenkoserhii** [comment]
> Сергію, додай себе як Assignee до пул реквесту і додай якийсь Label 

Done

File: src/components/custom/property/components/InfoBlock.tsx (hunk 1 above)
Comment (rudenkoserhii) [reply src/components/custom/property/components/InfoBlock.tsx:16]: fixed

File: src/components/custom/property/components/InfoBlock.tsx (hunk 2 above)
Comment (rudenkoserhii) [reply src/components/custom/property/components/InfoBlock.tsx:70]: fixed

File: src/components/custom/property/index.tsx (hunk 3 above)
Comment (rudenkoserhii) [reply src/components/custom/property/index.tsx:21]: fixed

File: src/components/custom/property/index.tsx (hunk 4 above)
Comment (rudenkoserhii) [reply src/components/custom/property/index.tsx:29]: fixed

**Review by anette1983 — APPROVED** [review]

File: src/components/Filter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+'use client';
+
+import { useTranslations } from 'next-intl';
+
+export default function Filter(): JSX.Element {
+  const t = useTranslations('Home');
+  return <p>Filter</p>;
```
Comment (lyudad) [inline src/components/Filter/index.tsx]: ?

File: src/components/custom/propertiesList/index.tsx [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+const DEFAULT_IMAGE = '/assets/images/home/properties_blank.jpg';
+
+function PropertiesList(): JSX.Element {
+  const [params, setParams] = useState<IPropertyParamsList>(INITIAL_PARAMS);
+  const { properties, propertiesError } = useGetProperties({ params: params });
+
+  const [propertiesList, setPropertiesList] = useState<IPropertyList>([]);
+  const [propertiesPagination, setPropertiesPagination] = useState<IPropertyPagination>();
+  const [error, setError] = useState<AxiosError | null>(null);
+  const [isFetching, setIsFetching] = useState(true);
```
Comment (lyudad) [inline src/components/custom/propertiesList/index.tsx]: types

File: src/components/custom/property/components/InfoBlock.tsx (hunk 1 above)
Comment (lyudad) [reply src/components/custom/property/components/InfoBlock.tsx:16]: добавьте это в правила притира, что бы мы больше не возвращались к этому вопросу

File: src/components/custom/property/components/SlickSlider.tsx [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+const MIN_SCALE = 1;
+const MAX_SCALE = 3;
+
+const SlickSlider: React.FC<SlickSliderProps> = ({ photos }) => {
+  const t = useTranslations('property');
+
+  const [toggleModal, setToggleModal] = useState<boolean>(false);
+  const [src, setSrc] = useState<string>('');
+  const [visibleArrows, setVisibleArrows] = useState<boolean>(false);
+  const [scale, setScale] = useState(MIN_SCALE);
```
Comment (lyudad) [inline src/components/custom/property/components/SlickSlider.tsx]: types

File: src/components/custom/property/components/ViewOnMap.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  coordinates: string | null | undefined;
+  closeModal: () => void;
+  openModal: boolean;
+}
+
+function ViewOnMap({ coordinates, closeModal, openModal }: ViewOnMapProps): JSX.Element {
+  const t = useTranslations('property');
+
+  const handleClose = (): void => closeModal();
+  const src = `https://maps.google.com/maps?q=${coordinates}&z=15&output=embed`;
```
Comment (lyudad) [inline src/components/custom/property/components/ViewOnMap.tsx]: линка должна быьт в константах

File: src/components/custom/property/index.tsx (hunk 3 above)
Comment (lyudad) [reply src/components/custom/property/index.tsx:21]: добавить в притир

File: src/components/custom/property/index.tsx (hunk 4 above)
Comment (lyudad) [reply src/components/custom/property/index.tsx:29]: проверить весь свой код а не только там где оставили коменты, так как пропущеных мест много

File: src/components/custom/property/index.tsx (hunk 3 above)
Comment (rudenkoserhii) [reply src/components/custom/property/index.tsx:21]: done

File: src/components/custom/property/index.tsx (hunk 4 above)
Comment (rudenkoserhii) [reply src/components/custom/property/index.tsx:29]: done

File: src/components/custom/property/components/SlickSlider.tsx (hunk 7 above)
Comment (rudenkoserhii) [reply src/components/custom/property/components/SlickSlider.tsx]: done

File: src/components/custom/propertiesList/index.tsx (hunk 6 above)
Comment (rudenkoserhii) [reply src/components/custom/propertiesList/index.tsx]: done

File: src/components/custom/property/components/ViewOnMap.tsx (hunk 8 above)
Comment (rudenkoserhii) [reply src/components/custom/property/components/ViewOnMap.tsx]: done

File: src/components/Filter/index.tsx (hunk 5 above)
Comment (rudenkoserhii) [reply src/components/Filter/index.tsx]: fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**